libs/wingc/src/type_check.rs

```diff
@@ -1244,61 +1288,61 @@ impl<'a> TypeChecker<'a> {
 		self.types.error()
 	}
 
-	fn make_error_variable_info(&self, is_static: bool) -> VariableInfo {
+	fn make_error_variable_info(&self) -> VariableInfo {
 		VariableInfo {
 			name: "<error>".into(),
 			type_: self.types.error(),
 			reassignable: false,
 			phase: Phase::Independent,
-			is_static,
-			is_member: false,
+			kind: VariableKind::Error,
 		}
 	}
 
 	// Validates types in the expression make sense and returns the expression's inferred type
-	fn type_check_exp(&mut self, exp: &Expr, env: &SymbolEnv) -> TypeRef {
+	fn type_check_exp(&mut self, exp: &Expr, env: &SymbolEnv) -> (TypeRef, Phase) {
 		CompilationContext::set(CompilationPhase::TypeChecking, &exp.span);
-		let t = self.type_check_exp_helper(&exp, env);
-		self.types.assign_type_to_expr(exp, t);
-		t
+		let (t, phase) = self.type_check_exp_helper(&exp, env);
+		self.types.assign_type_to_expr(exp, t, phase);
+		(t, phase)
 	}
 
 	/// Helper function for type_check_exp. This is needed because we want to be able to `return`
 	/// and break early, while still setting the evaluated type on the expression.
 	///
 	/// Do not use this function directly, use `type_check_exp` instead.
-	fn type_check_exp_helper(&mut self, exp: &Expr, env: &SymbolEnv) -> TypeRef {
+	fn type_check_exp_helper(&mut self, exp: &Expr, env: &SymbolEnv) -> (TypeRef, Phase) {
 		match &exp.kind {
 			ExprKind::Literal(lit) => match lit {
-				Literal::String(_) => self.types.string(),
-				Literal::Nil => self.types.nil(),
+				Literal::String(_) => (self.types.string(), Phase::Independent),
+				Literal::Nil => (self.types.nil(), Phase::Independent),
 				Literal::InterpolatedString(s) => {
 					s.parts.iter().for_each(|part| {
 						if let InterpolatedStringPart::Expr(interpolated_expr) = part {
-							let exp_type = self.type_check_exp(interpolated_expr, env);
+							let (exp_type, _) = self.type_check_exp(interpolated_expr, env);
 							self.validate_type_in(exp_type, &self.types.stringables(), interpolated_expr);
 						}
 					});
-					self.types.string()
+					// TODO: do we need to determine the phase based on the inner expressions?
+					(self.types.string(), Phase::Independent)
```

Comment:
I think as long as strings are phase-independent, it seems okay to me to say that the phase of the template string is independent

Reply:
OK.